### PR TITLE
cli: add git related variables in .gitlab-ci.yml

### LINKED ISSUE #259: set CI variables
There are a few variables that should be added for fine-tuning the gitlab-ci builds:

```yaml
variables:
  GIT_STRATEGY: clone
  GIT_SSL_NO_VERIFY: "true"
  GIT_SUBMODULE_STRATEGY: recursive
  GIT_LFS_SKIP_SMUDGE: 1
```

Once this is done, the gitlab-ci in `renku-ui` should also be changed.

### PR DESCRIPTION
**Pull request by jirikuncar**
(closes #259)

### CODE REVIEW

File: renku/cli/.gitlab-ci.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
-# renku: v1.dev20180619
+# renku: v1.dev20180723
+
+variables:
+  GIT_STRATEGY: clone
+  GIT_SSL_NO_VERIFY: "true"
```
Comment (rokroskar) [inline renku/cli/.gitlab-ci.yml]: not sure the quotes are needed?

**Comment by leafty** [comment]
Is it ok to see if fetch works fine?

File: renku/cli/.gitlab-ci.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
-# renku: v1.dev20180619
+# renku: v1.dev20180723
+
+variables:
+  GIT_STRATEGY: clone
```
Comment (rokroskar) [inline renku/cli/.gitlab-ci.yml]: @leafty is probably right, this should be changed to fetch

**Comment by rokroskar** [comment]
@leafty yes, you're right if we skip smudge the fetch should be sufficient

File: renku/cli/.gitlab-ci.yml (hunk 1 above)
Comment (jirikuncar) [reply renku/cli/.gitlab-ci.yml]: yes, because we want string "true" not YAML boolean `true`.

File: renku/cli/.gitlab-ci.yml (hunk 2 above)
Comment (jirikuncar) [reply renku/cli/.gitlab-ci.yml]: changed

**Comment by rokroskar** [comment]
@leafty you've tested this setup on staging?

**Comment by leafty** [comment]
I tried on internal.renku.ch with just
```yaml
variables:
  GIT_SSL_NO_VERIFY: "true"
```

**Comment by rokroskar** [comment]
Ok, but without `GIT_LFS_SKIP_SMUDGE: 1` you'll still pull from LFS, no?

**Comment by leafty** [comment]
Sure, but with the fetch strategy only the changed files are downloaded, not all of them.

**Pull request merged**